src/Avalonia.X11/X11TrayIconImpl.cs

```diff
@@ -0,0 +1,374 @@
+        public event Action OnTooltipChanged;
+        public Action<INativeMenuExporter> SetNativeMenuExporter { get; set; }
+        public Action<string> NewStatusAsync { get; set; }
+        public Action ActivationDelegate { get; set; }
+        public ObjectPath ObjectPath { get; }
+
+        public StatusNotifierItemDbusObj(ObjectPath dbusmenuPath)
+        {
+            ObjectPath = new ObjectPath($"/StatusNotifierItem");
+
+            _backingProperties = new StatusNotifierItemProperties
+            {
+                Menu = dbusmenuPath, // Needs a dbus menu somehow
+                ToolTip = new ToolTip("")
+            };
+
+            InvalidateAll();
+        }
+
+        public async Task ContextMenuAsync(int X, int Y)
+        {
+        }
+
+        public async Task ActivateAsync(int X, int Y)
+        {
+            ActivationDelegate?.Invoke();
+        }
+
+        public async Task SecondaryActivateAsync(int X, int Y)
+        {
+        }
+
+        public async Task ScrollAsync(int Delta, string Orientation)
+        {
+        }
+
+        public void InvalidateAll()
+        {
+            OnTitleChanged?.Invoke();
+            OnIconChanged?.Invoke();
+            OnOverlayIconChanged?.Invoke();
+            OnAttentionIconChanged?.Invoke();
+            OnTooltipChanged?.Invoke();
+        }
+
+        public async Task<IDisposable> WatchNewTitleAsync(Action handler, Action<Exception> onError = null)
+        {
+            OnTitleChanged += handler;
+            return Disposable.Create(() => OnTitleChanged -= handler);
+        }
+
+        public async Task<IDisposable> WatchNewIconAsync(Action handler, Action<Exception> onError = null)
+        {
+            OnIconChanged += handler;
+            return Disposable.Create(() => OnIconChanged -= handler);
+        }
+
+        public async Task<IDisposable> WatchNewAttentionIconAsync(Action handler, Action<Exception> onError = null)
+        {
+            OnAttentionIconChanged += handler;
+            return Disposable.Create(() => OnAttentionIconChanged -= handler);
+        }
+
+        public async Task<IDisposable> WatchNewOverlayIconAsync(Action handler, Action<Exception> onError = null)
+        {
+            OnOverlayIconChanged += handler;
+            return Disposable.Create(() => OnOverlayIconChanged -= handler);
+        }
+
+        public async Task<IDisposable> WatchNewToolTipAsync(Action handler, Action<Exception> onError = null)
+        {
+            OnTooltipChanged += handler;
+            return Disposable.Create(() => OnTooltipChanged -= handler);
+        }
+
+        public async Task<IDisposable> WatchNewStatusAsync(Action<string> handler, Action<Exception> onError = null)
+        {
+            NewStatusAsync += handler;
+            return Disposable.Create(() => NewStatusAsync -= handler);
+        }
+
+        public async Task<object> GetAsync(string prop)
+        {
+            if (prop.Contains("Menu"))
```

Comment:
This will return a value for `ThisStringContainsMenuSomewhereInIt` property which is invalid behavior

Reply:
this was a debug code, i'll just stub it out since it works without that anyway

Reply:
was removed.